tephra-core/src/main/java/org/apache/tephra/coprocessor/TransactionStateCache.java

```diff
@@ -78,6 +78,7 @@ protected void startUp() throws Exception {
   protected void shutDown() throws Exception {
     if (refreshService != null) {
       refreshService.interrupt();
+      refreshService.join(1000);
```

Comment:
`Thread.join(long millis)` does not throw an exception on timeout being reached. It only throws `IllegalArgumentException` when timeout is negative or `InterruptedException`.

Are you talking about handling the `InterruptedException`?

Reply:
I was mistaken, thinking that it would throw if the timeout is exceeded. Anyway, in a shutdown sequence we should catch all exceptions and ensure to continue the shutdown.